### PR TITLE
Send ical events if section interviewer changes

### LINKED ISSUE #461: Send ical events if section interviewer changes
And maybe if section canceled also

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #461

## PR includes

-   [x] Feature

### CODE REVIEW

File: src/modules/emaiMethods.ts [hunk 1]
```diff
@@ -1,11 +1,17 @@
+import { ICalCalendarMethod } from 'ical-generator';
+import RRule from 'rrule';
```
Comment (tsumo) [inline src/modules/emaiMethods.ts:2]: Default imports from rrule for some reason are always broken on my machine. Only `import { Rrule }` works

**Review by tsumo — APPROVED** [review]

File: src/modules/emaiMethods.ts [hunk 2]
```diff
@@ -51,15 +57,147 @@ export const cancelSectionEmail = async (id: number) => {
...
+        const { eventDetails, ...series } = await calendarMethods.getEventById(restException.eventId);
+
+        const exclude = [
+            ...series.exceptions
+                .map((exception) => exception.originalDate)
+                .filter((date) => date.toTimeString() !== originalDate.toTimeString()),
+            ...series.cancellations.map((cancelletion) => cancelletion.originalDate),
+        ];
+
+        console.log('exclude', exclude);
```
Comment (9teen90nine) [inline src/modules/emaiMethods.ts:88]: you left debug

File: src/modules/emailMethods.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            text: '',
+            icalEvent: calendarEvents({
+                method: ICalCalendarMethod.REQUEST,
+                events: [icalEventDataUpdateEvent],
+            }),
+        });
+    }
+
+    return sendMail({
+        from: 'Taskany Hire',
```
Comment (9teen90nine) [inline src/modules/emailMethods.ts:114]: you have copypaste 'from' and other fields. Make a wrapper function with default values.

File: src/modules/emailMethods.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+            sectionId,
+        })}`,
+        icalEvent: calendarEvents({
+            method: ICalCalendarMethod.REQUEST,
+            events: [icalEventDataException],
+        }),
+    });
+
+    await sendMail({
+        from: 'Hire',
```
Comment (9teen90nine) [inline src/modules/emailMethods.ts:185]: here too

**Review by 9teen90nine — CHANGES_REQUESTED** [review]

File: src/modules/nodemailer.ts [hunk 5]
```diff
@@ -13,15 +13,15 @@ export const transporter = nodemailer.createTransport({
...
     to: string;
     subject: string;
     text: string;
+    from?: string;
     html?: string;
     icalEvent?: string;
 };
 
-const message = ({ from, to, subject, text, html, icalEvent }: MessageBody) => ({
+const message = ({ from = 'Hire', to, subject, text, html, icalEvent }: MessageBody) => ({
```
Comment (9teen90nine) [inline src/modules/nodemailer.ts:24]: Taskany Hire?

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**